### ISSUE TITLE
Waiting list: Voucher redeemed, order cancelled / refunded - quota not changed

### ISSUE THREAD

**Issue by TimmThaler** [pretix/pretix#1405]
Hey, me again.

I just tested the waiting list with a test event.
* Quota of 12 tickets, 1 ticket already sold.
* Reduced the ticket quota (from 12 to 1) so that I could enter the waiting list.
* Entered the waiting list.
* Increased the quota (back to 12).
* Got the mail with the voucher code.
* Redeemed the voucher for the *one* ticket and completed checkout (available tickets now: 10).
* Got the mail with the confirmation.
* Cancelled the order.

Now the order was cancelled correctly, but the quota didn't increase by the 1 ticket I cancelled in the order, the number of available tickets still is at 10.

Perhaps I am too fast with my issue, but I thought, when a voucher is redeemed (doesn't show up in the backend at vouchers or waiting list) but the order cancelled again, the number of available tickets should increase again to 11 tickets. Instead one ticket is still shown as reserved for waiting list / vouchers.

:smiley:

**Comment by TimmThaler** [comment]
Hi Michael,

I've found out something new.

The voucher is automatically created from the waiting list entry and is valid for the time specified in the settings (default: 48hrs). In this period of time the voucher seems to be usable only once at a time, but re-usable, if you cancel a previous order using this voucher code.

After cancelling an order created by using a (waiting list) voucher, the counter at "payed orders" is not reduced by i.e. 1 (as my previous posts show). I think, after cancelling an order containing vouchers the counter of "payed orders" in a quota should be reduced and the counter for "waiting list (reserved)" should be increased again. This would be quite more transparent.

I just tested this with the quota shown above and everything works fine now.


---

**tl;dr**
Vouchers created from the waiting list are valid for the number of hours specified in the settings. In this time you can re-use a voucher, when an order using this voucher already is cancelled. After the specified time everything is fine again.

**Comment by raphaelm** [comment]
> After cancelling an order created by using a (waiting list) voucher, the counter at "payed orders" is not reduced by i.e. 1 (as my previous posts show). I think, after cancelling an order containing vouchers the counter of "payed orders" in a quota should be reduced and the counter for "waiting list (reserved)" should be increased again. This would be quite more transparent.

I'm having trouble to believe this. Is this happening on pretix Hosted or on a self-hosted instance? In the former case, can you send me details which events/orders this affected via email to support@pretix.eu, in the latter case, can you send me some before/after screenshots?

**Comment by TimmThaler** [comment]
It happens on pretix Hosted.
I'm going to test it again and send you some screenshots of the process :)
